core/io/file_access_encrypted.cpp

```diff
@@ -100,7 +100,7 @@ Error FileAccessEncrypted::open_and_parse(FileAccess *p_base, const Vector<uint8
 		MD5Update(&md5, (uint8_t *)data.ptr(), data.size());
 		MD5Final(&md5);
 
-		ERR_FAIL_COND_V(String::md5(md5.digest) != String::md5(md5d), ERR_FILE_CORRUPT);
+		ERR_FAIL_COND_V(String::md5(md5.digest) != String::md5(md5d), ERR_INVALID_ENCRYPTION_KEY);
```

Comment:
IMO it could just return `ERR_INVALID_DATA` so that we don't have to add a specific error code just for this use case.